packages/std/src/errors/std_error.rs

```diff
@@ -0,0 +1,313 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use snafu::Snafu;
+
+/// Structured error type for init, handle and query. This cannot be serialized to JSON, such that
+/// it is only available within the contract and its unit tests.
+///
+/// The prefix "Std" means "the standard error within the standard library". This is not the only
+/// result/error type in cosmwasm-std.
+#[derive(Debug, Serialize, Deserialize, Snafu, JsonSchema)]
+#[snafu(visibility = "pub(crate)")]
+#[serde(rename_all = "snake_case")]
+#[non_exhaustive]
```

Comment:
What does this mean? Can you send me a link? It seems interesting

Reply:
This allows the enum to declare that new cases are expected to be added in the future and the cases known right now are not exhaustive. Since this is interpreted by the compiler as part of the language, a `match` statement that handles all currently known cases is not exhaustive as long as there is no default case and compilations fails.
https://doc.rust-lang.org/1.35.0/unstable-book/language-features/non-exhaustive.html

Reply:
Thanks for the links.